src/dtype.rs

```diff
@@ -94,48 +120,60 @@ impl PyArrayDescr {
 
     /// Returns `self` as `*mut PyArray_Descr`.
     pub fn as_dtype_ptr(&self) -> *mut PyArray_Descr {
-        self.as_ptr() as _
+        self.as_borrowed().as_dtype_ptr()
     }
 
     /// Returns `self` as `*mut PyArray_Descr` while increasing the reference count.
     ///
     /// Useful in cases where the descriptor is stolen by the API.
     pub fn into_dtype_ptr(&self) -> *mut PyArray_Descr {
-        self.into_ptr() as _
+        (*self.as_borrowed()).clone().into_dtype_ptr()
```

Comment:
I am surprised at the need to explicit dereference here. Maybe this can be clarified by an additional let binding if the single-expression version is fragile?

Reply:
The problem is that we want to clone the `Bound` and not the `Borrowed`.
```suggestion
        let bound = &*self.as_borrowed();
        bound.clone().into_dtype_ptr()
```
I can do this if you prefer that. Maybe we should add `Borrowed::to_owned` to the public API in `pyo3` to make this use case nicer?

Reply:
> I can do this if you prefer that. Maybe we should add Borrowed::to_owned to the public API in pyo3 to make this use case nicer?

Yes, I think this would be nice. We still should not emphasize `Borrowed` to much. But I think this is a bit unwieldy as it stands.

Of course, here targeting the beta release, let's go for your suggestion.

Reply:
I've applied the suggestion and added a TODO to change to `Borrowed::to_owned` once available